### PR TITLE
ARROW-1347: [JAVA] return consistent child field name for List vectors

### PR DESCRIPTION
**Pull request by StevenMPhillips**

### CODE REVIEW

File: java/vector/src/main/java/org/apache/arrow/vector/complex/ListVector.java [hunk 1]
```diff
@@ -297,6 +298,9 @@ public int getBufferSize() {
 
   @Override
   public Field getField() {
+    if (getDataVector() == ZeroVector.INSTANCE) {
```
Comment (BryanCutler) [inline java/vector/src/main/java/org/apache/arrow/vector/complex/ListVector.java]: I know that the `vector` is initialized to `ZeroVector.INSTANCE` right now, but would this be a little better as  `if (getDataVector() instanceof ZeroVector)`?

**Review by BryanCutler — APPROVED** [review]
Just one small question, otherwise LGTM

File: java/vector/src/main/java/org/apache/arrow/vector/complex/ListVector.java (hunk 1 above)
Comment (jacques-n) [reply java/vector/src/main/java/org/apache/arrow/vector/complex/ListVector.java]: What was the previous field name?

Comment (BryanCutler) [reply java/vector/src/main/java/org/apache/arrow/vector/complex/ListVector.java]: I believe it was the `ZeroVector.name` which is "[DEFAULT]"

**Comment by wesm** [comment]
Can this be merged?

**Comment by jacques-n** [comment]
LGTM +1.

**Comment by BryanCutler** [comment]
Wouldn't it be better to use `instanceof`? I could change that and add a test for this if @StevenMPhillips is busy

**Comment by StevenMPhillips** [comment]
Thanks, @BryanCutler, if you want to go ahead and post your suggested change, I can merge it.

**Comment by wesm** [comment]
@StevenMPhillips can you close this stale PR?

**Pull request closed without merging**